src/lib/tecs/TECS.cpp

```diff
@@ -162,23 +165,32 @@ void TECSReferenceModel::update(const float dt, const AltitudeReferenceState &se
 	_alt_control_traj_generator.setMaxAccel(param.vert_accel_limit);
 	_alt_control_traj_generator.setMaxVel(fmax(param.max_climb_rate, param.max_sink_rate));
 
-	if (altitude_control_enable) {
-		const float target_climbrate = math::min(param.target_climbrate, param.max_climb_rate);
-		const float target_sinkrate = math::min(param.target_sinkrate, param.max_sink_rate);
-
-		const float delta_trajectory_to_target_m = setpoint.alt - _alt_control_traj_generator.getCurrentPosition();
-
-		float altitude_rate_target = math::signNoZero<float>(delta_trajectory_to_target_m) *
-					     math::trajectory::computeMaxSpeedFromDistance(
-						     param.jerk_max, param.vert_accel_limit, fabsf(delta_trajectory_to_target_m), 0.0f);
+	const float target_climbrate = math::min(param.target_climbrate, param.max_climb_rate);
+	const float target_sinkrate = math::min(param.target_sinkrate, param.max_sink_rate);
 
-		altitude_rate_target = math::constrain(altitude_rate_target, -target_sinkrate, target_climbrate);
+	if (height_rate_controlled) {
+		const float altitude_rate_target = math::constrain(setpoint.alt_rate, -target_sinkrate, target_climbrate);
 
 		_alt_control_traj_generator.updateDurations(altitude_rate_target);
 		_alt_control_traj_generator.updateTraj(dt);
```

Comment:
Maybe a stupid question, but Why do we still want to update the trajectory generator if we are running height rate control?

Reply:
"height rate controlled" is actually the wrong wording. We always control altitude, just that sometimes the interface to TECS is directly the altitude setpoint (eg in Auto flight), and sometimes its the height rate setpoint (ManualPosition, ManualAltitude, Descend).
We should have the discussion next if we can
- only have height rate as interface to TECS
- have the altitude reference model outside of TECS
- have the outer loop for altitude (generate height rate from the current altitude error) outside of TECS

Reply:
> only have height rate as interface to TECS
have the altitude reference model outside of TECS
have the outer loop for altitude (generate height rate from the current altitude error) outside of TECS

yes to all of the above. see comment here: https://github.com/PX4/PX4-Autopilot/pull/21189#issuecomment-1450541349